internal/test/wsu/wsu_test.go

```diff
@@ -98,32 +98,45 @@ func testAllVMs(t *testing.T) {
 func setupAndTest(t *testing.T, vmNum int) {
 	// Indicate that we can run the test suite on each node in parallel
 	t.Parallel()
-
+	vm := framework.WinVMs[vmNum]
 	// Run the WSU against the VM
-	wsuOut, err := runWSU(framework.WinVMs[vmNum])
-	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, wsuOut)
-
+	wsuOut, err := runWSU(vm)
+	wsuStringOutput := string(wsuOut)
+	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, wsuStringOutput)
+	// TODO: Think of a better way to refactor this function later to get just output that can be consumed by framework.
+	logFileName := "wsu.log"
```

Comment:
Won't this result in the first run getting overwritten? This function is run twice per VM. You need to differentiate between the two runs. You can use the test name for that.

Reply:
You're right. I did not think of that case. I'll update the log file's name to include the test name

Reply:
Update: There is a bug in the current code where we're not running tests twice on each node. @sebsoto will handle it in a separate PR. I'll add the test name.